db/blip_handler.go

```diff
@@ -477,7 +477,20 @@ func (bh *blipHandler) handleChanges(rq *blip.Message) error {
 		if nWritten > 0 {
 			output.Write([]byte(","))
 		}
-		if missing == nil {
+
+		deletedFlags := changesDeletedFlag(0)
+		// TODO: Add check for Blip Protocol Version when CBG-1435 is in
+		if len(change) > 3 {
+			var err error
+			deletedIntFlag, err := change[3].(json.Number).Int64()
+			if err != nil {
+				base.InfofCtx(bh.loggingCtx, base.KeyAll, "Failed to parse deletedFlags: %v", err)
+				continue
+			}
+			deletedFlags = changesDeletedFlag(deletedIntFlag)
```

Comment:
I know you have a TODO for protocol version, but wondering if it's worth doing a type switch instead, to avoid specific reliance on the protocol version.

```suggestion
			switch v := change[3].(type) {
				case json.Number:
					deletedIntFlag, err := v.Int64()
					if err != nil {
						base.InfofCtx(bh.loggingCtx, base.KeyAll, "Failed to parse deletedFlags: %v", err)
						return fmt.Errorf("Failed to parse deletedFlags: %v", err)
					}
					deletedFlags = changesDeletedFlag(deletedIntFlag)
				case bool:
					deletedFlags = changesDeletedFlagDeleted
				default:
					base.InfofCtx(bh.loggingCtx, base.KeyAll, "Unknown type for deleted field in changes message: %T", v)
					return fmt.Errorf("Unknown type for deleted field in changes message: %T", v)
			}
```

Reply:
Yeah, thats a good plan, will implement this. I don't think we want to return an error though in the error case. I think we want to 'continue' so we continue iterating over the other changes messages.